### ISSUE TITLE
How to configure the settings file for the DYS389II locus?

### ISSUE THREAD

**Issue by siyao** [Jerrythafast/FDSTools#3]
Hi Jerry，

I am using fdstools v2.1.0, and while analyzing the DYS389II locus, I set up flank and repeat parameters. However, the analysis results are mixed up with DYS389I, making it impossible to identify the true genotyping results. Additionally, the N48 in the DYS389 sequence structure [TAGA]a [CAGA]b N48[TAGA]c[CAGA]d was not recognized. Below are my library file, input.ini and the result files.
[file.zip](https://github.com/user-attachments/files/15862929/file.zip)

I am using the command `fdstools pipeline input.ini`.

Could you offer some advice on how to best configure the settings for the DYS389II locus? Thank you.

Siyao

**Comment by Jerrythafast** [comment]
Hi Siyao,

Short answer: the solution is to remove the `[repeat]` definition for DYS389II from your library file. FDSTools will then use ATAG and ACAG to shorten the repeats, therefore I recommend to also adjust the `[genome_position]` to target these repeats. In fact, this is everything you need in your library file (the other sections are unnecessary):
```ini
[genome_position]
DYS389I  = Y,12500447,12500494
DYS389II = Y,12500447,12500610
```

Long answer and explanation:
For almost all STR targets on the human genome (including DYS389) FDSTools (technically, STRNaming) has a built-in repeat definition, so you usually don't need to use the `[repeat]` section in your library file. This built-in bracketing mechanism is recommended by the DNA commission of the ISFG in [this paper](https://doi.org/10.1016/j.fsigen.2023.102946). The `[repeat]` section should be used to provide a repeat definition for a non-human target or for those rare instances where no built-in definition exists yet. The only current limitation is that the `[repeat]` section does not recognize repeat interruptions of more than 20 nucleotides (like the N48 in DYS389II). The `[flanks]` and `[block_length]` sections exist for similar reasons.

Kind regards,
Jerry

**Comment by siyao** [comment]
Okay, I see. Another question, if I provide the flanking sequence, fdstools can directly detect the SNPs in the flanking sequence?

**Comment by Jerrythafast** [comment]
Yes, that's also perfectly possible. For example, the [built-in library](https://github.com/Jerrythafast/FDSTools/blob/main/fdstools/data/libraries/ForenSeqA.ini) for the ForenSeq DNA Signature Prep kit contains these somewhat larger ranges so that it can also detect and report sequence variation outside of the main `ATAG[n]ACAG[m]` repeat:
```ini
[genome_position]
DYS389I     =  Y,  12500387,  12500513
DYS389II    =  Y,  12500436,  12500633
```

It will apply bracketing to some additional (secondary) repeats in this locus, but any sequence variation beyond that will be reported as SNP calls that look like `+12A>G`.

Jerry

**Comment by siyao** [comment]
Jerry，thank you for your answer.
